tests/lib/tools/tests.invariant

```diff
@@ -199,12 +199,47 @@ check_segmentation_violations() {
 	fi
 }
 
-check_fakestore_cleaned() {
-    # Check if fakestore was properly cleaned to avoid leaking into other tests.
-    if [ -f "/etc/systemd/system/snapd.service.d/store.conf" ]; then
-        echo "/etc/systemd/system/snapd.service.d/store.conf was not cleaned properly"
-        exit 1
-    fi
+# Check that there is only configuration files left are created by the test
+# harness (0X-xyz.conf) since others might override its values and affect tests
+check_snapd_svc_conf() {
```

Comment:
I am not sure if we need an invariant to check files in /etc/systemd/system/snapd.service.d/

in https://github.com/snapcore/snapd/blob/master/tests/lib/state.sh#L103 we restore the initial state (https://github.com/snapcore/snapd/blob/master/tests/lib/state.sh#L57) of that directory on each test. This is just done for classic.

Perhaps we can sync about this.

Reply:
can we do it on core then too? if not I think we should keep the invariant check